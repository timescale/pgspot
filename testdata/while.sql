create or replace function w1() returns void as $func$
declare
	_count bigint;
begin
	select 100 into strict _count;
	while format1('%s','false')::bool loop
	end loop;
	while 0 < _count loop
		_count := _count - 1;
    PERFORM format2('%s','true');
	end loop;
end
$func$ language plpgsql;
