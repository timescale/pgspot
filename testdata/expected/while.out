PS002: Unsafe function creation: w1() at line 1
PS005: Function without explicit search_path: w1() at line 1
PS017: Unqualified object reference: bool in CAST(format1('%s', 'false') AS bool) at line 1
PS016: Unqualified function call: format1 at line 1
PS001: Unqualified operator: '<' in 0 < _count at line 1
PS001: Unqualified operator: '-' in _count - 1 at line 1
PS016: Unqualified function call: format2 at line 1

 Errors: 1 Warnings: 6 Unknown: 0 

